updater/lib/dependabot/updater/group_update_creation.rb

```diff
@@ -232,6 +238,27 @@ def peer_dependency_should_update_instead?(dependency_name, dependency_files, up
               can_update?(requirements_to_unlock: :own)
           end
       end
+
+      def prepare_workspace
+        return unless job.clone? && job.repo_contents_path
```

Comment:
Can you explain why these checks are here? I looked at [job.clone?](https://github.com/dependabot/dependabot-core/blob/c9b38ced0d36e66339e92852030a1f9f9750b0ce/updater/lib/dependabot/job.rb#L122-L125) and it would be true if `vendor_dependencies?` is true. 

I think that we would want to use a workspace when `vendor_dependencies?` is true though?

Reply:
I think the intent of the checks is to make sure we only utilize the workspace in cases where we know a git repo is available. Since it's a `return unless`, I think it's doing what we want.

Reply:
We generally must clone as a precondition for `vendor_dependencies: true` to the best of my knowledge since we are manipulating content that is beyond the scope of the `FileFetcher` involved - there are some languages in which we always clone for this reason so the workspace will be invoked for those as well, but that's relatively useful for the extra debugging information even if we do extract/inject only manifest changes.

Over time we should look at expanding this so we always use the workspace for grouped updates so we have access to per-set diff - and the potential to experiment with extracting individual commits to open PRs with them instead of just the aggregate diff.